### PR TITLE
Speed up simplemap_mux by 9.6x.

### PR DESCRIPTION
**Pull request by rmlarsen**
This is accomplished by directly inserting the cell source attribute in the new object's 'attributes' map instead of creating a new attribute pool and then copying that. Based on a suggestion by Martin Poviser in a comment on https://github.com/YosysHQ/yosys/pull/3959

Topdown view before:
image

Topdown view after:
image

### CODE REVIEW

**Comment by rmlarsen** [comment]
Formatting PR in https://github.com/YosysHQ/yosys/pull/3969

**Comment by Ravenslofty** [comment]
I'm inclined to believe this should not be reformatted; it's a two line patch with 250 lines of unrelated changes, which makes it much more difficult to review.

Perhaps we can start by splitting "the patch" as a commit from the reformatting.

**Comment by povik** [comment]
> It's a two line patch with 250 lines of unrelated changes, which makes it much more difficult to review.

FWIW it's more than a two-line patch since Rasmus seems to have gone through other occurrences of the same pattern. Of course that not being obvious is even more of an argument for splitting the formatting from other changes.

Rasmus, I understand there may be frustration from on one hand being told not to include any extra whitespace changes in your patches and on the other hand seeing pushback on #3969. I feel like how we approach formatting is something that would benefit from a bit of discussion among Yosys maintainers.

**Comment by povik** [comment]
Not sure what the CI failure is about:
```
 make[1]: *** [macc.sh] Error 255
```
Let's see if I can trigger a re-run.

**Comment by rmlarsen** [comment]
> > It's a two line patch with 250 lines of unrelated changes, which makes it much more difficult to review.
> 
> FWIW it's more than a two-line patch since Rasmus seems to have gone through other occurrences of the same pattern. Of course that not being obvious is even more of an argument for splitting the formatting from other changes.
> 
> Rasmus, I understand there may be frustration from on one hand being told not to include any extra whitespace changes in your patches and on the other hand seeing pushback on #3969. I feel like how we approach formatting is something that would benefit from a bit of discussion among Yosys maintainers.

Yup, I'd appreciate some clear guidance, once way or the other.

**Comment by rmlarsen** [comment]
FWIW, contributing to Yosys is nowhere close to my day job, so I am trying to find the most friction-less way to land these improvements.

**Comment by rmlarsen** [comment]
Let me back the formatting changes out of this PR for now.

**Comment by povik** [comment]
So it seems the CI failure is not a fluke. I wonder what could have changed to be at fault here.

**Comment by rmlarsen** [comment]
@rmlarsen let's see if it still fails after I roll back the formatting changes. Hang on...

**Comment by rmlarsen** [comment]
@povik PTAL

**Comment by rmlarsen** [comment]
Hmm the test still fails on masOC. I'm afraid I don't understand how to debug this properly.

**Comment by whitequark** [comment]
Also on Linux, with a similar error: https://github.com/YosysHQ/yosys/actions/runs/6356727207/job/17266798637

**Comment by rmlarsen** [comment]
@whitequark thanks. I'm able to replicate here, after installing iverilog.

**Comment by rmlarsen** [comment]
Reverted changes to techmap.cc, which had no significant impact on timing anyway. PTAL.

**Comment by rmlarsen** [comment]
OK, it did have a small impact on timing. The speedup drops from 11.6x to 9.6x. I can live with that. :-)

**Review by Ravenslofty — APPROVED** [review]
I'm happy to merge this if someone else is.

**Comment by povik** [comment]
Go ahead!

**Pull request merged**
